src/Dropdown/Dropdown.js

```diff
@@ -1,189 +1,216 @@
-
-  state = {
-    isOpen: false,
-  };
+  // Wait for next tick after open to prevent page scroll when focusing
+  useEffect(() => {
+    if (isOpen) {
+      setTimeout(() => {
+        menuRef.current.focus();
+      });
+    }
+  }, [isOpen]);
 
-  componentWillUnmount() {
-    if (this.positioner) {
-      this.positioner.destroy();
+  useKeyPress('Escape', () => {
+    if (isOpen) {
+      close();
     }
-  }
+  });
 
-  componentDidUpdate() {
-    if (this.positioner && this.positioner.update) {
-      this.positioner.update();
+  const handleMenuBlur = () => {
+    if (autoclose) {
+      // Use timeout to delay examination of activeElement until after blur/focus
+      // events have been processed.
+      setTimeout(() => {
+        const nextActiveElement = document.activeElement;
+        if (!menuRef.current.contains(nextActiveElement)) {
+          close();
+        }
+      });
     }
-  }
+  };
 
-  show = () => {
-    const { placement, positionFixed, boundariesElement, offset } = this.props;
+  const renderFn = render || children;
 
-    this.setState(
-      {
-        isOpen: true,
-      },
-      () => {
-        this.positioner = new Popper(this.triggerRef.current, this.menuRef.current, {
-          placement,
-          positionFixed,
-          modifiers: {
-            offset: {
-              offset,
-            },
-            flip: {
-              behavior: ['left', 'bottom', 'top', 'right'],
-            },
-            preventOverflow: {
-              boundariesElement,
+  return (
+    <Manager>
+      <Reference>
+        {({ ref: triggerRef }) =>
+          React.cloneElement(trigger, {
+            ref: node => {
+              triggerRef(node);
+
+              const { ref } = trigger;
+              if (typeof ref === 'function') {
+                ref(node);
+              }
             },
-          },
-        });
-      }
-    );
-  };
+            ariahaspopup: true,
+            'aria-expanded': isOpen,
+            onClick: handleTrigger,
+          })
+        }
+      </Reference>
 
-  hide = () => {
-    this.setState(
-      {
-        isOpen: false,
-      },
-      () => {
-        this.positioner.destroy();
-      }
-    );
-  };
+      {isOpen && (
+        <Portal>
+          <Popper
+            innerRef={popperRef}
+            placement={placement}
+            positionFixed={positionFixed}
+            modifiers={{
+              offset: {
+                offset,
+              },
+              flip: {
+                enabled: false,
+              },
+              preventOverflow: {
+                boundariesElement,
+                padding: 12,
+              },
+              // Prevents wacky jumps in Transition component
+              computeStyle: { gpuAcceleration: false },
+            }}>
+            {({ ref, style }) => (
+              <Transition in={isOpen} timeout={0} appear>
+                {state => (
+                  <FocusTrap autoFocus={false}>
```

Comment:
Woot! This is a big accessibility win!

In the future, as mentioned in [the dropdown ticket](https://app.asana.com/0/821652619613450/1110934258569822/f), the defined behavior for dropdowns is to be cycled through via arrow keys, toggle via enter, and tab should continue to cycle through all other tabbale elements on the page (I had to build a vanilla html css custom dropdown in the past, which is why this is seared in my memory), as demonstrated here: https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html, and as enumerated here: https://www.w3.org/TR/wai-aria-practices-1.1/#listbox_kbd_interaction.

Downshift provides support for the specified behaviors, should we consider using that in the future to handle all of this?

Reply:
That's a great example but I'd argue our Dropdown is providing different functionality than what looks to be more of custom "select" box. The user engaging in an active action rather than choosing a value. The listbox above is most likely part of a form, which I believe is why tab would move to the next element or input. Conceptually, I think the dropdown is most like a local modal, so I think we'd want to abide by the criteria laid out in https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal 

As precedent, I think Twitter is one of the most accessible big box sites and I know they've poured a grand amount of time, attention and money into their accessbility. Below is an example of tabbing and arrowing through a dropdown.
![Kapture 2019-03-13 at 12 07 03](https://user-images.githubusercontent.com/14184138/54307312-976eca80-4588-11e9-906b-e858517be63b.gif)

Reply:
Thanks Kyle, you're right! A nav menu is a different set of criteria from a listbox.
Here are the [accessibility specifications](https://www.w3.org/TR/wai-aria-practices/#menu):
[An example](https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html#)
[Another example](https://www.w3.org/WAI/tutorials/menus/application-menus-code/)

Reply:
It seems that those examples don't allow tabbing to any element in the menu. I prefer to follow Google/Twitter/Slack precedent which allows tabbing to dropdown/popover items and keep tabbing/arrowing locked within the box.